auto_nag/scripts/fuzzing_bisection_without_regressed_by.py

```diff
@@ -2,41 +2,87 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from libmozdata.bugzilla import Bugzilla
+import re
 
-from auto_nag import utils
+import requests
+from libmozdata.bugzilla import Bugzilla, BugzillaUser
+
+from auto_nag import logger, utils
 from auto_nag.bzcleaner import BzCleaner
+from auto_nag.people import People
+
+PUSHLOG_PAT = re.compile(r"Pushlog: (.+)")
+BUG_PAT = re.compile(r"[\t ]*[Bb][Uu][Gg][\t ]*([0-9]+)")
+
+
+def is_ignorable_path(path: str) -> bool:
+    # TODO: also ignore other kinds of files that certainly can't cause regressions.
+
+    if any(
+        path.endswith(ext) for ext in (".txt", ".md", ".rst", ".pdf", ".doc", ".otf")
+    ):
+        return True
+
+    # This code was adapted from https://github.com/mozsearch/mozsearch/blob/2e24a308bf66b4c149683bfeb4ceeea3b250009a/router/router.py#L127
+    if (
+        "/test/" in path
+        or "/tests/" in path
+        or "/mochitest/" in path
+        or "/unit/" in path
+        or "/gtest/" in path
+        or "testing/" in path
+        or "/jsapi-tests/" in path
+        or "/reftests/" in path
+        or "/reftest/" in path
+        or "/crashtests/" in path
+        or "/crashtest/" in path
+        or "/gtests/" in path
+        or "/googletest/" in path
+    ):
+        return True
+
+    return False
 
 
 class FuzzingBisectionWithoutRegressedBy(BzCleaner):
+    def __init__(self) -> None:
+        super().__init__()
+        self.autofix_regressed_by = {}
+        self.bzmail_to_nickname = {}
+
     def description(self):
         return "Bugs with a fuzzing bisection and without regressed_by"
 
-    def get_max_ni(self):
-        return utils.get_config(self.name(), "max_ni")
-
     def handle_bug(self, bug, data):
         bugid = str(bug["id"])
         data[bugid] = {
             "assigned_to_email": bug["assigned_to"],
             "assigned_to_nickname": bug["assigned_to_detail"]["nick"],
-            "depends_on": bug["depends_on"],
         }
+        self.bzmail_to_nickname[bug["assigned_to"]] = bug["assigned_to_detail"]["nick"]
         return bug
 
     def set_autofix(self, bugs):
         for bugid, info in bugs.items():
-            self.add_auto_ni(
-                bugid,
-                {
-                    "mail": info["assigned_to_email"],
-                    "nickname": info["assigned_to_nickname"],
-                },
-            )
+            if "regressor_bug_id" in info:
+                self.autofix_regressed_by[bugid] = {
+                    "comment": {
+                        "body": "Setting regressed_by field after analyzing regression range found by bugmon."
+                    },
+                    "regressed_by": info["regressor_bug_id"],
+                }
+            elif "needinfo_target" in info:
+                self.add_auto_ni(
+                    bugid,
+                    info["needinfo_target"],
+                )
+
```

Comment:
Nit: free testing :)
```suggestion
            else:
                raise Exception("The bug should have regressor_bug_id or needinfo_target")

```

Reply:
Done in 4952d94.